ext/app/electron/config.ts

```diff
@@ -0,0 +1,206 @@
+          process.env[envKey] = defaultValue
+        } else {
+          process.env[envKey] = confValue
+        }
+      }
+    } else {
+      // envvar is defined, ignore config
+      if (confValue !== undefined) {
+        log.warn(`${confKey} is overridden by $${envKey}`)
+      }
+      if (!validator(envValue)) {
+        log.warn(`$${envKey} has invalid value ${envValue}, using default value ${defaultValue}`)
+        process.env[envKey] = defaultValue
+      }
+    }
+  }
+}
+
+
+function loadConfigFile(filename: string = DEFAULT_CONFIG_FILE) {
+  let config: Config
+  try {
+    let configBuffer = fse.readFileSync(filename)
+    config = new Config(ini.parse(configBuffer.toString("utf8")))
+  } catch (err) {
+    log.warn(`Failed to read configuration file: ${err}`)
+    config = new Config({})
+  }
+  // Section: login
+  config.apply(
+    "login.username",
+    "GRIST_DEFAULT_USERNAME",
+    NO_VALIDATION,
+    getUsername()
+  )
+  config.apply(
+    "login.email",
+    "GRIST_DEFAULT_EMAIL",
+    NO_VALIDATION,
+    getEmail()
+  )
+  // Section: server
+  config.apply(
+    "server.listen",
+    "GRIST_HOST",
+    NO_VALIDATION,
+    "localhost"
+  )
+  config.apply(
+    "server.port",
+    "GRIST_PORT",
+    (portstr) => {
+      if (! /^\d+$/.test(portstr)) {
+        return false
+      }
+      let port = parseInt(portstr)
+      return port > 0 && port < 65536
+    },
+    "_RANDOM"
+  )
+  config.apply(
+    "server.auth",
+    "GRIST_DESKTOP_AUTH",
+    (auth) => ["strict", "none", "mixed"].includes(auth),
+    "strict"
+  )
+  // Section: sandbox
+  config.apply(
+    "sandbox.flavor",
+    "GRIST_SANDBOX_FLAVOR",
+    (flavor) => ["pyodide", "unsandboxed", "gvisor", "macSandboxExec"].includes(flavor),
+    "pyodide"
+  )
+  // Section: storage
+  config.apply(
+    "storage.instance",
+    "GRIST_INST_DIR",
+    NO_VALIDATION,
+    APPDATA_DIR
+  )
+  config.apply(
+    "storage.documents",
+    "GRIST_DATA_DIR",
+    NO_VALIDATION,
+    electron.app.getPath("documents")
+  )
+  config.apply(
+    "storage.plugins",
+    "GRIST_USER_ROOT",
+    NO_VALIDATION,
+    path.join(electron.app.getPath("home"), ".grist")
+  )
+  config.apply(
+    "storage.homedb",
+    "TYPEORM_DATABASE",
+    NO_VALIDATION,
+    path.join(APPDATA_DIR, "home.sqlite3")
+  )
+
+  const homedb_location = path.parse((process.env.TYPEORM_DATABASE as string)).dir
+  if (!fse.existsSync(homedb_location)) {
+    log.warn(`Directory to contain the home DB does not exist, creating ${homedb_location}`)
+    fse.mkdirSync(homedb_location)
+  }
+
+  // We don't allow manually setting these envvars anymore. Fixing them makes maintaining grist-desktop easier.
+  process.env.GRIST_SINGLE_PORT = "true"
+  process.env.GRIST_SERVE_SAME_ORIGIN = "true"
+  process.env.GRIST_DEFAULT_PRODUCT = "Free"
+  process.env.GRIST_ORG_IN_PATH = "true";
+  process.env.GRIST_HIDE_UI_ELEMENTS = "helpCenter,billing,templates,multiSite,multiAccounts";
```

Comment:
This gives me some pause - but yes, probably for the best.

Reply:
Changing these doesn't really make sense for Desktop, and they are not documented anyway, so should be safe?

Reply:
Well, all these are documented in grist-core for those who understand the relationship - but yes, fine.

Reply:
Really wanna break the user impression that Desktop is simply a wrapper around Core. Hacky use of Desktop is difficult to officially support.